DataFormats/Provenance/src/ProductRegistry.cc

```diff
@@ -139,9 +139,9 @@ namespace edm {
 
   std::shared_ptr<ProductHolderIndexHelper> const&
```

Comment:
Actually, std::shared_ptr<ProductHolderIndexHelper const> is not sufficient because we also need the non-const case. std::shared_ptr<ProductHolderIndexHelper const> would work but requires a lot of changes.
Best I think is an overload set of two functions, a const function returning std::shared_ptr<ProductHolderIndexHelper const>, and a non-const finction returning std::shared_ptr<ProductHolderIndexHelper>. Already coded and working.